### DISCUSSION TITLE
is there documentation of what this default function App() does? in example basic root https://github.com/remix-run/examples/blob/main/basic/app/root.tsx

### DISCUSSION THREAD

**Discussion by putuoka** [remix-run/remix#5676] [Q&A] [answered]
No other file call this App() function so it must be build in and i can't find the documentation or explanation

**Comment by kiliman** [comment, marked as answer]
The *root* route is the tip of your route tree. Remix will render this route and any child routes.

Like any Remix route, the `default` export is the UI component. And, like any default export, you can give it an optional name. The name itself doesn't matter.

The convention is to name it `App`, but it can be named `Root` or `MyApp`, etc.
```ts
export default function MyApp() {}
```
https://remix.run/docs/en/main/file-conventions/root

**Reply by putuoka** [reply]
> The _root_ route is the tip of your route tree. Remix will render this route and any child routes.
> 
> Like any Remix route, the `default` export is the UI component. And, like any default export, you can give it an optional name. The name itself doesn't matter.
> 
> The convention is to name it `App`, but it can be named `Root` or `MyApp`, etc.
> 
> ```ts
> export default function MyApp() {}
> ```
> 
> https://remix.run/docs/en/main/file-conventions/root

Thank you for answering. if in react or react native export default MyApp() this function should be called to use it that's why i;m dizzy the function name never call but it executed lol

**Reply by kiliman** [reply]
Yes, Remix will always call the default export of your routes automatically as it renders the UI. The root route is the only *special* route that must exist.
